crates/phactory/src/prpc_service.rs

```diff
@@ -1097,13 +1086,13 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> PhactoryApi for Rpc
             return Err(from_display("Invalid challenge"));
         }
         // 3. verify sgx local attestation report to ensure the handover pRuntimes are on the same machine
-        if !dev_mode {
+        if !dev_mode && attestation_provider != "" && attestation_provider != "none" {
```

Comment:
So many `if !dev_mode && attestation_provider != "" && attestation_provider != "none" {`.
It's not very clear what the conditions indicate. Is it possible just to be `if !dev_mode {` or `if !dev_mode && in_sgx {`.

Reply:
changed as your suggests